--- UsernamePanel.js	2022-03-02 15:36:23.799648816 +0530
+++ UsernamePanel-new-password.js	2022-03-02 15:37:03.193416459 +0530
@@ -60,12 +60,28 @@
 		var panel = new Ext.Panel({
 			id: 'usernameCheck',
 			bbar: [
-				this.forgotBtn = new Ext.Button({
+			/*	this.forgotBtn = new Ext.Button({
 					cls: "go-login-forgot-username",
 					text: t("Forgot login credentials?"),
 					handler: this.showForgot,
 					scope: this
-				}),
+				}), */
+
+this.changePassButton = new Ext.Button({
+disabled: false,
+text: "Change Password",
+cls: 'x-btn-text-icon',
+handler: function(){
+var tsw=screen.width-20;
+var tsh=screen.width-20;
+tLeftPosition = (screen.width) ? (screen.width-tsw)/2 : 0;
+tTopPosition = (screen.height) ? (screen.height-tsh)/2 : 0;
+teailarchivenewwindow=window.open('/changepass/','changepasswin','height='+tsh+',width='+tsw+',top='+tTopPosition+',left='+tLeftPosition+',scrollbars=yes,menubar=yes,resizable'); if (window.focus) {teailarchivenewwindow.focus()}
+
+},
+scope: this
+}),
+
 				'->',
 				this.nextButton = new Ext.Button({
 					type: "submit",
